apps/rush-lib/src/cli/actions/CheckAction.ts

```diff
@@ -27,14 +27,20 @@ export default class CheckAction extends BaseRushAction {
   }
 
   protected run(): Promise<void> {
-    const pinnedVersions: { [dependency: string]: string } = {};
-    this.rushConfiguration.pinnedVersions.forEach((version: string, dependency: string) => {
-      pinnedVersions[dependency] = version;
+    // Collect all the preferred versions into a single table
+    const allPreferredVersions: { [dependency: string]: string } = {};
+
+    this.rushConfiguration.commonVersions.preferredVersions.forEach((version: string, dependency: string) => {
+      allPreferredVersions[dependency] = version;
+    });
+
+    this.rushConfiguration.commonVersions.xstitchPreferredVersions.forEach((version: string, dependency: string) => {
+      allPreferredVersions[dependency] = version;
     });
```

Comment:
Maybe put this in the commonVersions class?

Reply:
I also added a MapExtensions.mergeFromMap() helper to node-core-library

---
In reply to: [178636369](https://github.com/Microsoft/web-build-tools/pull/593#discussion_r178636369) [](ancestors = 178636369)